test/python/circuit/test_parameters.py

```diff
@@ -1782,6 +1782,184 @@ def test_parameter_symbol_equal_after_ufunc(self):
         self.assertEqual(phi._parameter_symbols, cos_phi._parameter_symbols)
 
 
+class TestParameterReferences(QiskitTestCase):
+    """Test the ParameterReferences class."""
+
+    def test_equal_inst_diff_instance(self):
+        """Different value equal instructions are treated as distinct."""
+        from qiskit.circuit.library.standard_gates.rz import RZGate
```

Comment:
There's usually no need to hide imports in tests, and that'll cut down the import repetition - if anything is shadowing the name `RZGate` in the test, we're already screwed haha.

Reply:
Heh. Good point. I just copied this from some other nearby tests, assuming they had a good reason for doing it.